models/stationModel.ts

```diff
@@ -0,0 +1,97 @@
+import { PoolClient } from "pg";
+import pool from "../config/db";
+
+interface Station {
+  id: string;
+  station_name: string;
+  location: string;
+  station_code: string;
+  updated_at: Date;
+}
+
+const createStation = async (
+  station_name: string,
+  location: string,
+  station_code: string
+): Promise<Station | null> => {
+  try {
+    const result = await pool.query(
+      `INSERT INTO stations(station_name, location, station_code) VALUES($1, $2, $3) RETURNING *`,
+      [station_name, location, station_code]
+    );
+    return result.rows[0] || null;

```

Comment:
you are returning NULL, instead of it you have to return blank array here. avoiding using NULL is a good practice